src/pages/admin/settings.js

```diff
@@ -0,0 +1,107 @@
+import { Typography, TextField, Button, Snackbar, Box } from "@mui/material";
+import { gql, useQuery, useMutation } from "@apollo/client";
+import { useState } from "react";
+
+const QUERY = gql`
+    query {
+        settings {
+            membershipRequirement
+        }
+    }
+`;
+
+const MUTATION = gql`
+	mutation ($membershipRequirement: Int!) {
+        alterSettings(membershipRequirement: $membershipRequirement) {
+            membershipRequirement
+        }
+	}
+`;
+
+const classes = {
+	mainDiv: {
+		width: "1200px",
+		maxWidth: "90%",
+		margin: "auto"
+	}
+};
+
+const ManageSettings = () => {
+    const {
+		_loading,
+		_error,
+		data
+	} = useQuery(QUERY);
+
+    let [settings, setSettings] = useState(null)
+    let [changed, setChanged] = useState(false)
+
+    const [save] = useMutation(MUTATION, {
+        onCompleted() {
+            setChanged(false);
+        }
+    });
+
+    if (data && settings === null) {
+        setSettings(data.settings)
+    }
+
+    const saveSettings = async () => {
+        const confirmation = window.confirm("Are you sure you want to update these settings?");
```

Comment:
May be better to use a [MUI alert](https://mui.com/material-ui/react-dialog/#alerts) for consistent visual style, and to prevent browser blocking (at least on Firefox, I get the option to disable these notices)